packages/@aws-cdk/aws-apigatewayv2/lib/common/domain-name.ts

```diff
@@ -57,14 +82,42 @@ export interface DomainNameProps {
    */
   readonly domainName: string;
   /**
-   * The ACM certificate for this domain name
+   * The ACM certificate for this domain name.
+   * Certificate can be both ACM issued or imported.
    */
   readonly certificate: ICertificate;
   /**
    * The mutual TLS authentication configuration for a custom domain name.
    * @default - mTLS is not configured.
    */
-  readonly mtls?: MTLSConfig
+  readonly mtls?: MTLSConfig;
+  /**
+   * The user-friendly name of the certificate that will be used by the endpoint for this domain name.
+   * This property is optional and is helpful if you have too many certificates and it is easier to remember
+   * certificates by some name rather that the domain they are valid for.
+   * @default - No friendly certificate name
+   */
+  readonly certificateName?: string;
+
+  /**
+   * The type of endpoint for this DomainName.
+   * @default EndpointType.REGIONAL
+   */
+  readonly endpointType?: EndpointType;
+
+  /**
+   * The Transport Layer Security (TLS) version + cipher suite for this domain name.
+   * @default SecurityPolicy.TLS_1_2
+   */
+  readonly securityPolicy?: SecurityPolicy;
+
+  /**
+   * A public certificate issued by ACM to validate that you own a custom domain. This parameter is required
+   * only when you configure mutual TLS authentication and you specify an ACM imported or private CA certificate
+   * for `certificate`. The ownership verification certificate validates that you have permissions to use the domain name.
+   * @default - only required when configuring mTLS
+   */
+  readonly ownershipVerificationCertificate?: ICertificate;
```

Comment:
Very long name. Can we make this terser, maybe `ownershipCertificate`?

Reply:
sure